python/lsst/daf/butler/_butler.py

```diff
@@ -1068,19 +1092,24 @@ def pruneCollection(self, name: str, purge: bool = False, unstore: bool = False)
             Raised if the butler is read-only or arguments are mutually
             inconsistent.
         """
+
         # See pruneDatasets comments for more information about the logic here;
         # the cases are almost the same, but here we can rely on Registry to
         # take care everything but Datastore deletion when we remove the
         # collection.
         if not self.isWriteable():
             raise TypeError("Butler is read-only.")
-        if purge and not unstore:
-            raise TypeError("Cannot pass purge=True without unstore=True.")
         collectionType = self.registry.getCollectionType(name)
+        if purge and not unstore:
+            raise PruneCollectionsArgsError(PruneCollectionsArgsError.Reason.PURGE_WITHOUT_UNSTORE,
```

Comment:
I'm fine with a new subclass for specificity but couldn't we also make three different exceptions? You could have a new base class of PruneCollectionsArgsError and then have three subclasses of that such as PurgeWithoutUnstoreCollectionsError etc. They could have the error message burned in and you wouldn't need the enum here and you wouldn't need to check the reason later on -- you could simply catch the specific exception and replace it with a new error message suitable for click users.